File: Backend/FPAMS/FPAMS.Domain/Entities/AcademicYear.cs
using FPAMS.Domain.Common;

namespace FPAMS.Domain.Entities;

public class AcademicYear : BaseEntity
{
    public string YearName { get; set; } = string.Empty;

    public DateTime StartDate { get; set; }

    public DateTime EndDate { get; set; }

    public bool IsCurrent { get; set; }

    public bool IsActive { get; set; } = true;
}

File: Frontend/fpams-web/src/types/academicYear.ts
export interface AcademicYear {
    id: string;
    yearName: string;
    startDate: string;
    endDate: string;
    isCurrent: boolean;
    isActive: boolean;
}

export interface CreateAcademicYearRequest {
    yearName: string;
    startDate: string;
    endDate: string;
    isCurrent: boolean;
}

export interface UpdateAcademicYearRequest {
    id: string;
    yearName: string;
    startDate: string;
    endDate: string;
    isCurrent: boolean;
    isActive: boolean;
}
